app/lib/git_repository.rb

```diff
@@ -38,20 +38,32 @@ def initialize(connect_string)
         @repos.reset('origin/master', :hard) # align to whatever is in origin/master
       rescue StandardError # TODO this shouldn't be necessary. It catches the case when a local repo is corrupted,
                            #      we need to prevent the corruption instead/as well
-        bad_repo_path = "#{connect_string}.bad"
-        FileUtils.rm_rf(bad_repo_path)
-        FileUtils.mv(connect_string, bad_repo_path)
-        repo_path, _sep, repo_name = connect_string.rpartition(File::SEPARATOR)
-        bare_path = File.join(repo_path, 'bare', "#{repo_name}.git")
-        @repos = Rugged::Repository.clone_at(bare_path, connect_string)
-        m_logger = MarkusLogger.instance
-        m_logger.log "Recloned corrupted git repo: #{connect_string}"
+        GitRepository.reclone_repo(connect_string)
       end
     else
-      raise "Repository does not exist at path \"#{@repos_path}\""
+      GitRepository.reclone_repo(connect_string)
     end
   end
 
+  def self.reclone_repo(connect_string)
+    repo_path, _sep, repo_name = connect_string.rpartition(File::SEPARATOR)
+    bare_path = File.join(repo_path, 'bare', "#{repo_name}.git")
+    raise "Cannot reclone from non-existant bare repository #{bare_path}" unless Dir.exist?(bare_path)
```

Comment:
Use the same message that is used in svn and memory, which should really be a single I18n entry in the future:
Respository does not exist